### PR TITLE
NETOBSERV-844 Unable to have a working statusUrl in FlowCollector with Loki Operator 5.6

### PR DESCRIPTION
**Pull request by jpinsonneau**
Added `statusTls` option in CRD for status URL

Check example yaml:
```yaml
    # Uncomment lines below for typical installation with loki-operator (5.6+ needed)
    # url: 'https://loki-gateway-http.netobserv.svc:8080/api/logs/v1/network/'
    # statusUrl: 'https://loki-query-frontend-http.netobserv.svc:3100/'
    # authToken: FORWARD
    # statusTls:
    #   enable: true
    #   caCert:
    #     certFile: service-ca.crt
    #     name: loki-ca-bundle
    #     type: configmap
    #   userCert:
    #     certFile: tls.crt
    #     certKey: tls.key
    #     name: loki-query-frontend-http
    #     type: secret
```

### CODE REVIEW

File: config/samples/flows_v1beta1_flowcollector.yaml [hunk 1]
```diff
@@ -68,7 +68,18 @@ spec:
     # Uncomment lines below for typical installation with loki-operator (5.6+ needed)
     # url: 'https://loki-gateway-http.netobserv.svc:8080/api/logs/v1/network/'
     # statusUrl: 'https://loki-query-frontend-http.netobserv.svc:3100/'
-    # authToken: HOST
+    # authToken: FORWARD
```
Comment (jpinsonneau) [inline config/samples/flows_v1beta1_flowcollector.yaml:71]: `FORWARD` is recommended in the doc; we should also recommend it in the sample

File: config/samples/flows_v1beta1_flowcollector.yaml [hunk 2]
```diff
@@ -68,7 +68,18 @@ spec:
...
+    #   enable: true
+    #   caCert:
+    #     certFile: service-ca.crt
+    #     name: loki-ca-bundle
+    #     type: configmap
+    #   userCert:
+    #     certFile: tls.crt
+    #     certKey: tls.key
+    #     name: loki-query-frontend-http
+    #     type: secret
```
Comment (jpinsonneau) [inline config/samples/flows_v1beta1_flowcollector.yaml:82]: `statusTls` needs both `loki-ca-bundle` and `loki-query-frontend-http` crt and key files for mTLS

File: controllers/consoleplugin/consoleplugin_objects.go [hunk 3]
```diff
@@ -184,6 +185,18 @@ func buildArgs(desired *flowslatest.FlowCollectorSpec) []string {
...
+	if statusTLS.Enable {
+		if statusTLS.InsecureSkipVerify {
+			args = append(args, "-loki-status-skip-tls")
+		} else {
+			args = append(args, "--loki-status-ca-path", helper.GetCACertPath(&statusTLS, lokiStatusCerts))
+			args = append(args, "--loki-status-user-cert-path", helper.GetUserCertPath(&statusTLS, lokiStatusCerts))
+			args = append(args, "--loki-status-user-key-path", helper.GetUserKeyPath(&statusTLS, lokiStatusCerts))
+		}
+	}
+
```
Comment (msherif1234) [inline controllers/consoleplugin/consoleplugin_objects.go:199]: can we extend unit-test to cover this newly added field ?

File: controllers/consoleplugin/consoleplugin_objects.go [hunk 4]
```diff
@@ -184,6 +185,18 @@ func buildArgs(desired *flowslatest.FlowCollectorSpec) []string {
 			args = append(args, "--loki-ca-path", helper.GetCACertPath(&desired.Loki.TLS, lokiCerts))
 		}
 	}
+
+	statusTLS := helper.LokiStatusTLS(&desired.Loki)
+	if statusTLS.Enable {
+		if statusTLS.InsecureSkipVerify {
+			args = append(args, "-loki-status-skip-tls")
```
Comment (msherif1234) [inline controllers/consoleplugin/consoleplugin_objects.go:192]: is this a typo it should be `--loki-status-skip-tls` ?

File: pkg/helper/flowcollector.go [hunk 5]
```diff
@@ -42,6 +42,13 @@ func LokiForwardUserToken(spec *flowslatest.FlowCollectorLoki) bool {
 	return spec.AuthToken == flowslatest.LokiAuthForwardUserToken
 }
 
+func LokiStatusTLS(spec *flowslatest.FlowCollectorLoki) flowslatest.ClientTLS {
```
Comment (msherif1234) [inline pkg/helper/flowcollector.go:45]: nit: `GetLokiStatusTLS` ?

File: controllers/consoleplugin/consoleplugin_objects.go (hunk 4 above)
Comment (jpinsonneau) [reply controllers/consoleplugin/consoleplugin_objects.go:192]: I think we should harmonize everything with a single minus character as it's already the case for most of these args. I will also fix the existing `--loki-ca-path`

Thanks for pointing that !

File: controllers/consoleplugin/consoleplugin_objects.go (hunk 3 above)
Comment (jpinsonneau) [reply controllers/consoleplugin/consoleplugin_objects.go:199]: https://github.com/netobserv/network-observability-operator/pull/307/commits/7d64e7c3fd3a2f9a6912dc138cc6307ea34c76bd#diff-22b7d1b615362ec58a371291942fa2bfe7290375729e4fb6b0be03f3af4ca993

File: pkg/helper/flowcollector.go [hunk 6]
```diff
@@ -42,6 +42,13 @@ func LokiForwardUserToken(spec *flowslatest.FlowCollectorLoki) bool {
...
+	if spec.StatusTLS != nil {
+		return *spec.StatusTLS
+	}
+	return spec.TLS
```
[the remark this answers is not in the document]
Comment (jpinsonneau) [reply pkg/helper/flowcollector.go:49]: I'm fine with that :+1:

[the remark this answers is not in the document]
Comment (jpinsonneau) [reply pkg/helper/flowcollector.go]: https://github.com/netobserv/network-observability-operator/pull/307/commits/5af9b70721815a8598e72441c412a1a1330229c3

File: api/v1beta1/flowcollector_types.go [hunk 7]
```diff
@@ -462,6 +462,7 @@ type FlowCollectorLoki struct {
 	// This is useful to show error messages and some context in the frontend.
 	// When using the Loki Operator, set it to the Loki HTTP query frontend service, for example
 	// https://loki-query-frontend-http.netobserv.svc:3100/.
+	// statusTLS configuration will be used when statusUrl is set.
```
Comment (msherif1234) [inline api/v1beta1/flowcollector_types.go:465]: // +optional ?

**Comment by jpinsonneau** [comment]
@Amoghrd I've restored the default values as we discussed together https://github.com/netobserv/network-observability-operator/pull/307/commits/467289f18f34111c09f37733856ccaf3d7d5472c

```
              "statusTls": {
                "caCert": {
                  "certFile": "service-ca.crt",
                  "name": "loki-ca-bundle",
                  "type": "configmap"
                },
                "userCert": {
                  "certFile": "tls.crt",
                  "certKey": "tls.key",
                  "name": "loki-query-frontend-http",
                  "type": "secret"
                },
                "enable": false,
                "insecureSkipVerify": false
              },
```

**Pull request merged**